Lib/Src/CUartCom.cpp

```diff
@@ -181,42 +196,25 @@ void CUartCom::endTx()
  */
 void CUartCom::uartRxHandler(UART_HandleTypeDef *p_huart)
 {
-    // Store incoming char
-    static uint8_t len_counter = 0;
+    // Store char in buffer
+    if (m_rx_buffer.put((char)m_rx_char) == false)
+    {
+        // Reset buffer and log error
+        send("Error: Max string size is RX_BUF_SIZE\n");
+        m_rx_buffer.reset();
+    }
 
-    bool full_buffer = (len_counter >= (RX_BUF_SIZE));
     if (m_rx_char == '\n' || m_rx_char == '\r')
     {
-        // '\n' and '\r' are replaced with '\0' to mark the end of the string
-        m_rx_buffer.put('\0');
-
-        if (full_buffer)
+        // Get string and store it in RX QUEUE
+        etl::string<MAX_STRING_SIZE> rx_string = getString();
+        if (m_rx_queue.size() <= MAX_RX_QUEUE_SIZE && !rx_string.empty())
         {
-            /**
-             * @note If a command length exceeds the RX_BUF_SIZE, The whole
-             * command will be ignored.
-             */
-
-            // Send warning
-            send("[ERROR]: Command exceeds max. length\n");
-
-            // Reset buffer
-            m_rx_buffer.reset();
-        }
-        else
-        {
-            etl::string<MAX_STRING_SIZE> rx_string = getString();
-            if (m_rx_queue.size() <= MAX_RX_QUEUE_SIZE && !rx_string.empty())
+            if (m_rx_queue.put(rx_string) == false)
             {
-                m_rx_queue.put(rx_string);
+                send("Error: Buffer overflow -> RX Queue\n");
```

Comment:
I would advocate we don't send this type of error back through the uart. I suggest we log it and then dump the log of errors when requested. Talk to @Masitech about how he organised logging in the past.